templates/terraform/nested_query.go.erb

```diff
@@ -34,21 +34,25 @@
 	<% else -%>
 		item := vRaw.(map[string]interface{})
 	<% end -%>
+		var expectedV, itemV interface{}
+	  var err error
+
 	<% object.identity.each do |prop| -%>
 	<% if settable_properties.include?(prop) -%>
-		itemIdV, err := expand<%= resource_name -%><%= titlelize_property(prop) -%>(d.Get("<%= prop.name.underscore -%>"), d, meta.(*Config))
+		expectedV, err = expand<%= resource_name -%><%= titlelize_property(prop) -%>(d.Get("<%= prop.name.underscore -%>"), d, meta.(*Config))
 		if err != nil {
 			return nil, err
 		}
 	<% else -%>
-		itemIdV := d.Get("<%= prop.name.underscore -%>")
+		expectedV = d.Get("<%= prop.name.underscore -%>")
 	<% end # settable_properties.include?(prop)-%>
-		actualIdV := flatten<%= resource_name -%><%= titlelize_property(prop) -%>(item["<%= prop.api_name %>"], d)
-		log.Printf("[DEBUG] Checking if item's <%= prop.api_name %> (%#v) is equal to resource's (%#v)", itemIdV, actualIdV)
-		if !reflect.DeepEqual(itemIdV, actualIdV) {
+		itemV = flatten<%= resource_name -%><%= titlelize_property(prop) -%>(item["<%= prop.api_name %>"], d)
+		log.Printf("[DEBUG] Checking if item's <%= prop.api_name %> (%#v) is equal to resource's (%#v)", itemV, expectedV)
+		if !reflect.DeepEqual(itemV, expectedV) {
 			continue
 		}
+
 	<% end # object.identity.each -%>
-		return item, nil
+		return item, err
```

Comment:
I don't totally understand what problem this is solving, can you clarify? `err` is only set on line 42, and it's immediately returned if it isn't nil, so at this point, it can only be nil, right?

Reply:
i can't initialize err without using it